students/seongsu/westagram/users/views.py

```diff
@@ -41,4 +41,26 @@ def post(self, request):
         
         except KeyError as e:
             return JsonResponse({"message" : "KEY_ERROR: " + str(e).replace("'", '')}, status = 400)
+
+class SignInView(View):
+    def post(self, request):
+        
+        user_data = json.loads(request.body)
+
+        try:
+            email    = user_data['email']
+            password = user_data['password']
+
+            if not User.objects.filter(email = email):
+                return JsonResponse({"message" : "invaild email"}, status = 400)
+            
+            elif User.objects.get(email = email).password != password:
```

Comment:
1. input으로 받은 email과 password로 매칭되는 유저가 존재하는지 한번의 쿼리로 확인할 수 있습니다.
2. 존재 여부만 True/False로 확인하는 경우 True/False를 반환하는 `exists()` 함수를 사용하는 것이 효율적이고 가독성 측면에서 도 좋습니다.

```suggestion
            if not User.objects.filter(email = email, password=password).eixsts():
                return JsonResponse({"message" : "invaild email"}, status = 400)
```

Reply:
수정했습니다